sdk/nodejs/server/server.ts

```diff
@@ -0,0 +1,131 @@
+import ts from "typescript"
+
+import Client from "../api/client.gen.js"
+import { UnknownDaggerError } from "../common/errors/UnknownDaggerError.js"
+import { connect } from "../connect.js"
+import { entrypoinsMetadatatoGQLSchema } from "./convertor.js"
+import { EntrypointMetadata } from "./entrypointMetadata.js"
+import { serializeSignature, serializeSymbol } from "./serialization.js"
+import { listFiles, writeFile, readFile } from "./utils.js"
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type Entrypoint = (client: Client, ...args: string[]) => any
+
+export type Resolver = {
+  name: string
+  namespace: string
+}
+
+export type Input = {
+  // Use type any until we find a way to dynamically asserts type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  args: { [key: string]: any }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  parent: { [key: string]: any }
+
+  resolver: string
+}
+
+export async function getSchema(...entrypoints: Entrypoint[]): Promise<void> {
+  const metadatas: EntrypointMetadata[] = []
+  const files = await listFiles()
```

Comment:
Do you think we need to get every file in the dir here, or can we just provide the file that has the `serveCommands` call (which should be in the callstack of this function)?

My guess was that `program.getSourceFiles` would then handle visiting every file imported by that one, but could be wrong.

Not a big deal, just a potential small simplification I am curious about.